Schulungsumgebung/playbook.yaml

```diff
@@ -0,0 +1,342 @@
+        src: "~/.ssh/id_rsa.pub"
+        dest: "/tmp/remote-{{ inventory_hostname }}-id_rsa.pub"
+        flat: true
+
+    - name: Allow remote user to login remote
+      tags: execute
+      ansible.posix.authorized_key:
+        user: "{{ linux_user }}"
+        state: present
+        key: "{{ lookup('file','/tmp/remote-{{inventory_hostname}}-id_rsa.pub')}}"
+
+    - name: Copy ssh-config to the webtop directory
+      tags: execute
+      ansible.builtin.copy:
+        src: .ssh
+        dest: webtop/
+        remote_src: true
+        mode: 0700
+
+    - name: Create ssh config file
+      tags: execute
+      ansible.builtin.copy:
+        dest: "webtop/.ssh/config"
+        mode: 0600
+        content: |
+          Host localhost
+            Hostname localhost
+            User "{{ linux_user }}"
+            StrictHostKeyChecking no
+
+    - name: Download JBoss EAP Zip
+      tags: execute
+      middleware_automation.redhat_csp_download.redhat_csp_download:
+        username: "{{ redhat_csp_username }}"
+        password: "{{ redhat_csp_password }}"
+        url: "https://access.redhat.com/jbossnetwork/restricted/softwareDownload.html?softwareId=99481"
+        dest: "Downloads/jboss-eap-{{ jboss_version }}.zip" # Ja hier muss man den Filenamen angeben.
+
+    - name: Clone Troubleshooting Workshop
+      tags: execute
+      ansible.builtin.git:
+        repo: https://github.com/Gepardec/TroubleshootingWorkshop.git
+        dest: "{{ ansible_env.HOME }}/TroubleshootingWorkshop"
+        version: master
+
+    - name: Copy Elastic APM File
+      tags: execute
+      ansible.builtin.template:
+        src: templates/docker-compose.yml
+        dest: elastic-apm/
+        mode: 0644
+
+    - name: Start Elastic APM
+      become: true
+      tags: execute
+      community.docker.docker_compose:
+        project_src: elastic-apm
+        state: present
+
+    - name: Download elastic-apm-agent
+      tags: execute
+      ansible.builtin.get_url:
+        url: https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/1.12.0/elastic-apm-agent-1.12.0.jar
+        dest: "{{ ansible_env.HOME }}/elastic-apm/elastic-apm-agent-1.12.0.jar"
+
+    - name: Download MAT
+      tags: execute
+      ansible.builtin.get_url:
+        url: https://www.eclipse.org/downloads/download.php?file=/mat/1.13.0/rcp/MemoryAnalyzer-1.13.0.20220615-linux.gtk.x86_64.zip&r=1
+        dest: "{{ ansible_env.HOME }}/Downloads/mat.zip"
+        timeout: 120
+
+    - name: Extract mat.zip
+      tags: execute
+      ansible.builtin.unarchive:
+        src: "{{ ansible_env.HOME }}/Downloads/mat.zip"
+        dest: "{{ ansible_env.HOME }}/"
+        remote_src: true
+
+    - name: Download JMC
+      tags: execute
+      ansible.builtin.get_url:
+        url: https://github.com/AdoptOpenJDK/openjdk-jmc-overrides/releases/download/8.1.0/org.openjdk.jmc-8.1.0-linux.gtk.x86_64.tar.gz
+        dest: "{{ ansible_env.HOME }}/Downloads/jmc.tar.gz"
+        timeout: 120
+
```

Comment:
Interessant, wusste nicht dass es JMC für OpenJDK gibt.

Reply:
Wurde Opensource gestellt. Angeblich kann man auch ein Flightrecord in openjdk 8 machen: https://developers.redhat.com/blog/2020/08/25/get-started-with-jdk-flight-recorder-in-openjdk-8u